skiko/src/iosMain/kotlin/org/jetbrains/skiko/SkikoUIView.kt

```diff
@@ -19,15 +19,52 @@ class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol, UITextPaste
 
     private var skiaLayer: SkiaLayer? = null
     private var _inputDelegate: UITextInputDelegateProtocol? = null
-    private var _pasteConfiguration: UIPasteConfiguration? = null
-    private var _pasteDelegate: UITextPasteDelegateProtocol? = null
+    private var _currentTextMenu:TextMenuArguments? = null
 
     constructor(skiaLayer: SkiaLayer, frame: CValue<CGRect> = CGRectNull.readValue()) : super(frame) {
         this.skiaLayer = skiaLayer
     }
 
+    fun showTextMenu(textMenu: TextMenuArguments) {
+        _currentTextMenu = textMenu
+        val menu: UIMenuController = UIMenuController.sharedMenuController()
+        if (menu.isMenuVisible()) {
+            menu.hideMenu()
+        }
+        val rect  = textMenu.targetRect
+        val cgRect = CGRectMake(
+            x = rect.left.toDouble(),
+            y = rect.top.toDouble(),
+            width = rect.width.toDouble(),
+            height = rect.height.toDouble()
+        )
+        menu.showMenuFromView(this, cgRect)
+    }
+
+    fun hideTextMenu() {
+        _currentTextMenu = null
```

Comment:
Currently show/hide behavior is controlled by common code (like in Android).
So, in Android text menu is not hidden from being tapping outside.

Maybe in future we can add logic to hide menu with tapping outside.

Reply:
> Currently show/hide behavior is controlled by common code

But here we show the native menu, and it can behave its own way, common logic can't intercept click outside, because currently it doesn't know about menu bounds. Did you check that it doesn't hide?

> Maybe in future we can add logic to hide menu with tapping outside.

We should stick to the native behavior here. If the native text menu behaves this way, we should eventually do the same. Could you check, how native text menus behave in native text fields?

Reply:
I will try to achieve native iOS behavior

Reply:
> Yeah, I checked on Compose - menu doesn't hide (like in Android).

I checked how it behaves on Android. It seems, that it doesn't hide when we tap on non-focusable component, but when we tap, for example, on another textfield, the tap is received by this textfield, and in result, focus changes, and in response to the change of focus - we hide the text menu.

> But on native iOS apps - it hides!

In that case, better to stick to hiding when we tap outside. Also we should investigate - should we consume the tap or send it to the tapped component. It can be done in another PR, if that will be more convenient. But I am afraid that we have to change some part of the public interfaces (but maybe not).

Reply:
I will investigate possible solution. And I will let you know!

Reply:
Unfortunately for now, it's hard to achieve. A lot of changes in common code of Compose.

Reply:
We decided to discuss it later